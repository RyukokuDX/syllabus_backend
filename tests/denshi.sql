-- 電子の科目で、教科書の料金が2000円以下（教科書なしも含む）、かつ選択科目の一覧
-- File Version: v1.0.0
-- Project Version: v2.2.3
-- Last Updated: 2025-07-02

SELECT DISTINCT
	cache_data->>'科目名' as 教科名,
	instructor->>'氏名' as 担当教員,
	book->>'書名' as 教科書名,
	book->>'価格' as 教科書価格
FROM syllabus_cache
	CROSS JOIN LATERAL (
		SELECT opening_info
		FROM jsonb_array_elements(cache_data->'開講情報一覧') as opening_info
		WHERE jsonb_typeof(opening_info) = 'object'
	) as opening
	CROSS JOIN LATERAL (
		SELECT syllabus
		FROM jsonb_array_elements(opening.opening_info->'シラバス一覧') as syllabus
		WHERE jsonb_typeof(syllabus) = 'object'
	) as syl
	CROSS JOIN LATERAL (
		SELECT instructor
		FROM jsonb_array_elements(syl.syllabus->'担当教員一覧') as instructor
		WHERE jsonb_typeof(instructor) = 'object'
	) as instructor
	LEFT JOIN LATERAL (
		SELECT book
		FROM jsonb_array_elements(syl.syllabus->'教科書一覧') as book
		WHERE jsonb_typeof(book) = 'object'
		LIMIT 1
	) as book ON TRUE
WHERE cache_name = 'subject_syllabus_cache'
  -- 履修要綱で電子の選択科目
  AND EXISTS (
    SELECT 1
    FROM jsonb_array_elements(cache_data->'履修情報一覧') as curriculum_info,
         jsonb_array_elements(curriculum_info->'履修要綱一覧') as requirement_info
    WHERE requirement_info->>'学部課程' LIKE '%電子%'
      AND requirement_info->>'必須度' = '選択'
  )
  -- 教科書が2000円以下または教科書が存在しない場合
  AND (
    book->>'価格' IS NULL
    OR
    (book->>'価格')::int <= 2000
  )
ORDER BY 教科名, 担当教員; 