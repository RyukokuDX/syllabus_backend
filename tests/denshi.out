SQLファイルを実行中: tests/denshi.sql
             教科名             | 担当教員  |                                教科書名                                | 教科書価格 
--------------------------------+-----------+------------------------------------------------------------------------+------------
 ASEANグロ-バルプログラム       | 奥田 哲士 |                                                                        | 
 ASEANグロ-バルプログラム       | 宮武 智弘 |                                                                        | 
 アルゴリズムとデ-タ構造I·演習  | 木村 昌弘 |                                                                        | 
 アルゴリズムとデ-タ構造II·演習 | 木村 昌弘 |                                                                        | 
 グロ-バル人材育成プログラム    | 富崎 欣也 |                                                                        | 
 ディジタル信号処理·演習        | 吉井 一倫 | ディジタル信号処理                                                     | 
 ディジタル論理                 | 小堀 聡   |                                                                        | 
 デザインシンキング             | 宮武 智弘 |                                                                        | 
 ネットワ-ク通信システム        | 張 毅波   |                                                                        | 
 プログラミング法I·演習         | 酒田 信親 |                                                                        | 
 プログラミング法II·演習        | 酒田 信親 |                                                                        | 
 プロジェクトリサ-チI           | 奥田 哲士 |                                                                        | 
 プロジェクトリサ-チI           | 吉井 一倫 |                                                                        | 
 プロジェクトリサ-チI           | 山本 哲男 |                                                                        | 
 プロジェクトリサ-チI           | 青井 芳史 |                                                                        | 
 プロジェクトリサ-チI           | 樋口 三郎 |                                                                        | 
 プロジェクトリサ-チI           | 野口 佳樹 |                                                                        | 
 プロジェクトリサ-チII          | 奥田 哲士 |                                                                        | 
 プロジェクトリサ-チII          | 吉井 一倫 |                                                                        | 
 プロジェクトリサ-チII          | 山本 哲男 |                                                                        | 
 プロジェクトリサ-チII          | 青井 芳史 |                                                                        | 
 プロジェクトリサ-チII          | 樋口 三郎 |                                                                        | 
 プロジェクトリサ-チII          | 野口 佳樹 |                                                                        | 
 応用プログラミング·演習        | 中川 晃成 |                                                                        | 
 回路設計·演習                  | 木村 睦   | 電子回路シミュレータPSpice入門編 : 電子回路の動作をパソコンで疑似体験! | 1890
 計算機制御·演習                | 酒田 信親 |                                                                        | 
 計測工学                       | 山本 伸一 |                                                                        | 
 情報数学                       | 木村 昌弘 |                                                                        | 
 人工知能                       | 小堀 聡   |                                                                        | 
 線形代数学II·演習              | 中川 晃成 |                                                                        | 
 微分積分学II·演習              | 中川 晃成 |                                                                        | 
 符号理論                       | 石崎 俊雄 |                                                                        | 
 物理·演習                      | 張 陽軍   |                                                                        | 
 理工学のすすめ                 | 越川 博元 |                                                                        | 
 理工学のすすめ                 | 岸本 圭子 |                                                                        | 
 理工学のすすめ                 | 宮武 智弘 |                                                                        | 
 理工学のすすめ                 | 小寺 康博 |                                                                        | 
 理工学のすすめ                 | 植田 祥明 |                                                                        | 
 理工学のすすめ                 | 大塩 裕哉 |                                                                        | 
 理工学のすすめ                 | 池田 聖   |                                                                        | 
 理工学のすすめ                 | 藤原 和将 |                                                                        | 
 理工学のすすめ                 | 樋口 三郎 |                                                                        | 
 理工学のすすめ                 | 富崎 欣也 |                                                                        | 
 理工学のすすめ                 | 木村 睦   |                                                                        | 
(44 行)

